Common/src/geometry_structure.cpp

```diff
@@ -2792,6 +2794,90 @@ void CGeometry::UpdateCustomBoundaryConditions(CGeometry **geometry_container, C
   }
 }
 
+
+void CGeometry::ComputeSurf_Straightness(CConfig *config, bool print_on_screen) {
+
+  bool RefUnitNormal_defined;
+  unsigned short iMarker,
+                 iDim;
+  unsigned long iVertex,
+                iPoint;
+  su2double epsilon = 1.0e-6,
+            Area;
+
+  su2double *Normal        = new su2double[nDim],
+            *UnitNormal    = new su2double[nDim],
+            *RefUnitNormal = new su2double[nDim];
+
+  /*--- Allocate memory and assume the less restrictive, i.e. all boundaries are not straight ---*/
+  bound_is_straight = new bool [nMarker];
+  for (iMarker = 0; iMarker < nMarker; iMarker++) 
+    bound_is_straight[iMarker] = false;
+
+  /*--- Loop over all the markers ---*/
+  for (iMarker = 0; iMarker < nMarker; iMarker++) {
+    
+    if (config->GetMarker_All_KindBC(iMarker) == SYMMETRY_PLANE ||
+        config->GetMarker_All_KindBC(iMarker) == EULER_WALL) {
+      
+      /*--- Assume now that this boundary marker is straight. As soon as one
+        AreaElement is found that is not aligend with a Reference then it is 
+        certain that the boundary marker is not straight and one can stop 
+        searching ---*/
+      bound_is_straight[iMarker] = true;
+      RefUnitNormal_defined = false; 
+      iVertex = 0;
+
+      while(bound_is_straight[iMarker] == true && iVertex < nVertex[iMarker]) {
+        
+        iPoint = vertex[iMarker][iVertex]->GetNode();
+        
+        if (node[iPoint]->GetDomain()) {
+          
+          vertex[iMarker][iVertex]->GetNormal(Normal);
+          /*--- Compute unit normal. ---*/
+          Area = 0.0;
+          for (iDim = 0; iDim < nDim; iDim++)
+            Area += Normal[iDim]*Normal[iDim];
+          Area = sqrt (Area);
+          
+          for (iDim = 0; iDim < nDim; iDim++)
+            UnitNormal[iDim] = -Normal[iDim]/Area;
+          
+          /*--- Check if unit normal is within tolerance of the Reference unit normal. 
+                Reference unit normal = first unit normal found. ---*/
+          if(RefUnitNormal_defined) {
+            for (iDim = 0; iDim < nDim; iDim++) {
+              if( abs(RefUnitNormal[iDim] - UnitNormal[iDim]) > epsilon )
```

Comment:
I think the advantage via *unit* normal is that it is independent of any 'real world problem' size. Therefore the chosen epsilon can stay easily constant.  Otherwise I guess some scaling of epsilon with problem dimension would be advisable.